src/Sylius/Component/Core/Promotion/Checker/NthOrderRuleChecker.php

```diff
@@ -57,6 +59,10 @@ public function isEligible(PromotionSubjectInterface $subject, array $configurat
             return false;
         }
 
+        if (null === $customer->getId()) {
+            return $this->orderRepository->countByCustomerEmailAndPaymentState($customer->getEmail(), PaymentInterface::STATE_COMPLETED) === ($configuration['nth'] - 1);
```

Comment:
This should be removed. When the customer is not in the database it means it is his first order. Proper implementation with explanatory comment:

```php
// On first guest order the customer is not yet saved to the storage
if (null === $customer->getId()) {
    return 1 === $configuration['nth'];
}
```